### PR TITLE
Doc changes for EGW local Preference

### PR DESCRIPTION
**Pull request by sridhartigera**
Product Version(s):

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 1]
```diff
@@ -229,6 +229,7 @@ an instance (for example when scaling up the cluster).
 - [Verify the feature operation](#verify-the-feature-operation)
 - [Control the use of egress gateways](#control-the-use-of-egress-gateways)
 - [Policy enforcement for flows via an egress gateway](#policy-enforcement-for-flows-via-an-egress-gateway)
+- [Prefer local egress gateways](#prefer-local-egress-gateways)
```
Comment (bmckercher123) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]:
```suggestion
- [Set local egress gateways as the default](#set-local-egress-gateways-as-the-default)
```

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 2]
```diff
@@ -1175,6 +1176,45 @@ patch release.
 
 :::
 
+## Prefer local egress gateways
```
Comment (bmckercher123) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]:
```suggestion
## Set local egress gateways as the default
```

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 3]
```diff
@@ -1175,6 +1176,45 @@ patch release.
...
+
+From v3.18, preference can be set to use the local egress gateways. `GatewayPreference` can be set to `PreferNodeLocal` or `None` in the Egress Gateway Policy. By default `GatewayPreference` is set to `None`.
```
Comment (bmckercher123) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]:
```suggestion
When you set the `GatewayPreference` to `PreferNodeLocal` or `None`, egress gateways local to the client's node are used if available. If there are no local egress gateways, the fallback is to use the other egress gateways. Default: `None`. 
```

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 4]
```diff
@@ -1175,6 +1176,45 @@ patch release.
...
+When set to `PreferNodeLocal`, egress gateways local to the client's node will be used if available. If there are no local egress gateways, we will fallback to using the other egress gateways.
```
Comment (bmckercher123) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]: Delete and combine with above paragraph.

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 5]
```diff
@@ -1175,6 +1176,45 @@ patch release.
...
+
+The following is an example of Egress Gateway Policy with GatewayPreference configured.
```
Comment (bmckercher123) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]:
```suggestion
The following example is an egress gateway policy with the `GatewayPreference` set to use local egress gateways. 
```

**Review by bmckercher123 — APPROVED** [review]
Suggestions for clarity

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx (hunk 2 above)
Comment (fasaxc) [reply calico-enterprise/networking/egress/egress-gateway-aws.mdx]: @bmckercher123 This suggestion differs from the one above (but they need to agree for the link to match), do you have a preference?

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx [hunk 6]
```diff
@@ -1175,6 +1176,45 @@ patch release.
...
+
+```yaml noValidation
+apiVersion: projectcalico.org/v3
+kind: EgressGatewayPolicy
+metadata:
+  name: "egw-policy1"
+spec:
+  rules:
+  // callout-for-next-line
```
Comment (fasaxc) [inline calico-enterprise/networking/egress/egress-gateway-aws.mdx]: What does this special comment do (my guess is that it makes a 1, 2, 3 appear int eh example so you can reference it below but I'm not sure you're doing that)?

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx (hunk 2 above)
Comment (bmckercher123) [reply calico-enterprise/networking/egress/egress-gateway-aws.mdx]: Yes, they should match. I missed the first instance. Updated to "Set"

**Review by bmckercher123 — CHANGES_REQUESTED** [review]
You'll need to copy these changes to CC 3.18 as well.

File: calico-enterprise/networking/egress/egress-gateway-aws.mdx (hunk 6 above)
Comment (sridhartigera) [reply calico-enterprise/networking/egress/egress-gateway-aws.mdx]: Yes right. This is not needed.

File: calico-cloud/networking/egress/egress-gateway-on-prem.mdx [hunk 7]
```diff
@@ -324,6 +328,15 @@ The health port `8080` is used by:
   no policy is required to allow it. {{prodname}} only sends probes to egress gateway pods that have a named
   "health" port. This ensures that during an upgrade, health probes are only sent to upgraded egress gateways.
 
+### Deploying on a RKE2 CIS Hardened Cluster
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-on-prem.mdx]:
```suggestion
### Deploy on a RKE2 CIS-hardened cluster
```

File: calico-cloud/networking/egress/egress-gateway-on-prem.mdx [hunk 8]
```diff
@@ -175,10 +177,12 @@ patch release.
 - [Enable egress gateway support](#enable-egress-gateway-support)
 - [Provision an egress IP pool](#provision-an-egress-ip-pool)
 - [Deploy a group of egress gateways](#deploy-a-group-of-egress-gateways)
+- [Configure iptables backend for egress gateways](#configure-iptables-backend-for-egress-gateways)
 - [Configuring namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-on-prem.mdx]:
```suggestion
- [Configure namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```

File: calico-cloud/networking/egress/egress-gateway-on-prem.mdx [hunk 9]
```diff
@@ -324,6 +328,15 @@ The health port `8080` is used by:
   no policy is required to allow it. {{prodname}} only sends probes to egress gateway pods that have a named
   "health" port. This ensures that during an upgrade, health probes are only sent to upgraded egress gateways.
 
+### Deploying on a RKE2 CIS Hardened Cluster
+
+If you are deploying `egress-gateway` on a RKE2 CIS-hardened cluster, its `PodSecurityPolicies` restrict the `securityContext` and `volumes` required by egress gateway. When deploying using the egress gateway custom resource, the Tigera Operator sets up `PodSecurityPolicy`, `Role`, `RoleBinding` and associated `ServiceAccount`.
+
+### Configure iptables backend for egress gateways
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-on-prem.mdx:335]:
```suggestion
#### Configure iptables backend for egress gateways
```

File: calico-cloud/networking/egress/egress-gateway-on-prem.mdx [hunk 10]
```diff
@@ -324,6 +328,15 @@ The health port `8080` is used by:
...
+
+The Tigera Operator configures egress gateways to use the same iptables backend as `calico-node`.
+To modify the iptables backend for egress gateways, you must change the `iptablesBackend` field in the [Felix configuration](../../reference/resources/felixconfig.mdx).
+
 ### Configuring namespaces and pods to use egress gateways
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-on-prem.mdx]:
```suggestion
### Configure namespaces and pods to use egress gateways
```

File: calico-cloud/networking/egress/egress-gateway-aws.mdx [hunk 11]
```diff
@@ -223,11 +223,13 @@ an instance (for example when scaling up the cluster).
 - [Enable AWS-backed IP pools](#enable-aws-backed-ip-pools)
 - [Configure IP pools backed by VPC subnets](#configure-ip-pools-backed-by-vpc-subnets)
 - [Deploy a group of egress gateways](#deploy-a-group-of-egress-gateways)
+- [Configure iptables backend for egress gateways](#configure-iptables-backend-for-egress-gateways)
 - [Configuring namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-aws.mdx]:
```suggestion
- [Configure namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```

File: calico-cloud/networking/egress/egress-gateway-aws.mdx [hunk 12]
```diff
@@ -852,6 +853,11 @@ EOF
 
 :::
 
+### Configure iptables backend for egress gateways
+
+The Tigera Operator configures egress gateways to use the same iptables backend as `calico-node`.
+To modify the iptables backend for egress gateways, you must change the `iptablesBackend` field in the [Felix configuration](../../reference/resources/felixconfig.mdx).
+
 ### Configuring namespaces and pods to use egress gateways
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-aws.mdx]:
```suggestion
### Configure namespaces and pods to use egress gateways
```

File: calico-cloud/networking/egress/egress-gateway-azure.mdx [hunk 13]
```diff
@@ -130,11 +130,13 @@ Azure Route Server deployment is per VNet, as such each VNet with egress gateway
 - [Configure route reflector](#configure-route-reflector)
 - [Enable egress gateway support](#enable-egress-gateway-support)
 - [Deploy a group of egress gateways](#deploy-a-group-of-egress-gateways)
+- [Configure iptables backend for egress gateways](#configure-iptables-backend-for-egress-gateways)
 - [Configuring namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```
Comment (bmckercher123) [inline calico-cloud/networking/egress/egress-gateway-azure.mdx]:
```suggestion
- [Configure namespaces and pods to use egress gateways](#configuring-namespaces-and-pods-to-use-egress-gateways)
```

**Review by bmckercher123 — APPROVED** [review]
A few more nits, then we can merge.

**Review by bmckercher123 — APPROVED** [review]
LGTM

**Pull request merged**